bin/restore_backup.py

```diff
@@ -0,0 +1,267 @@
+    for table in tables:
+        name = table.split('.', 1)[0]
+        pipeline.add_s3_bucket(name + "Bucket", s3_backup + "/DDB/" + table)
+        pipeline.add_ddb_table(name, table)
+        pipeline.add_emr_copy(name+"Copy",
+                              Ref(name + "Bucket"),
+                              Ref(name),
+                              Ref("RestoreCluster"),
+                              export=False)
+
+    for table in tables:
+        name = table.split('.', 1)[0]
+        resp = input("Delete existing data in {}? [y/N] ".format(name))
+        if resp and len(resp) > 0 and resp[0].lower() == 'y':
+            ddb_delete_data(session, table)
+
+    return pipeline
+
+def rds_pipeline(session, domain, directory, component, rds_name):
+    names = AWSNames(domain)
+    # XXX: AZ `E` is incompatible with T2.Micro instances (used by backup)
+    subnet = aws.subnet_id_lookup(session, 'b-' + names.internal)
+
+    s3_backup = "s3://backup." + domain + "/" + directory
+    s3_log = "s3://backup." + domain + "/restore-logs/"
+
+
+    _, data = list_s3_bucket(session, "backup." + domain, directory + "/RDS/" + rds_name)
+    if len(data) == 0:
+        print("No {} table data backed up on {}, skipping restore".format(component, directory))
+        return None
+
+    pipeline = DataPipeline(fmt="DP", log_uri = s3_log, resource_role="backup")
+    pipeline.add_shell_command("RDSRestore",
+                               "bash ~/rds.sh restore {}".format(rds_name),
+                               source = Ref("RDSBucket"),
+                               runs_on = Ref("RDSInstance"))
+    pipeline.add_ec2_instance("RDSInstance",
+                              subnet = subnet,
+                              image = aws.ami_lookup(session, "backup.boss")[0])
+    pipeline.add_s3_bucket("RDSBucket", s3_backup + "/RDS/" + rds_name)
+
+    return pipeline
+
+def endpoint_rds_pipeline(session, domain, directory):
+    names = AWSNames(domain)
+
+    return rds_pipeline(session,
+                        domain,
+                        directory,
+                        "Endpoint",
+                        names.endpoint_db)
+
+def auth_rds_pipeline(session, domain, directory):
+    names = AWSNames(domain)
+
+    return rds_pipeline(session,
+                        domain,
+                        directory,
+                        "Auth",
+                        names.auth_db)
+
+if __name__ == '__main__':
+    types = ['consul', 'vault', 'dynamo', 'endpoint', 'auth']
+    parser = argparse.ArgumentParser(description = "Script the restoration of a backup")
+    parser.add_argument("--aws-credentials", "-a",
+                        metavar = "<file>",
+                        default = os.environ.get("AWS_CREDENTIALS"),
+                        type = argparse.FileType('r'),
+                        help = "File with credentials to use when connecting to AWS (default: AWS_CREDENTIALS)")
+    parser.add_argument("--ami-version",
```

Comment:
Is this the AMI the pipeline(s) will run on?

Reply:
Like `bin/cloudformation.py` the `--ami-version` flag can be used to select the BOSS AMI version that is used. The custom AMI is used for backup and restore of non DynamoDB data.